hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestAHSv2ClientImpl.java

```diff
@@ -107,6 +113,91 @@ public void testGetAppAttemptReport() throws IOException, YarnException {
         isEqualTo("test original tracking url");
   }
 
+  @Test
+  public void testGetContainerByAppAttempt() throws IOException, YarnException {
+    // Single container tests
+    final ApplicationId appId1 = ApplicationId.newInstance(0, 1);
+    final ApplicationId appId2 = ApplicationId.newInstance(0, 2);
```

Comment:
Is the second application necessary? It tests the same thing, just with a different ID.

Reply:
The logic behind is to prove that an application won't appear in a wrong place.